honeycombio/honeycomb-opentelemetry-web
# feat: Add LOAF support for onINP

## Which problem is this PR solving?

- Closes #138

## Short description of the changes

Optionally add performance timing as spans for `INP` core web vital if `includeTimingsAsSpans` is set to true.

```js
webVitalsInstrumentationConfig: {
      vitalsToTrack: ['CLS', 'FCP', 'FID', 'INP', 'LCP', 'TTFB'],
      inp: {
        reportAllChanges: true,
        includeTimingsAsSpans: true,
      },
...
```

## How to verify that this has the expected result
To force an poor INP you can add a snippet like this:
```js
const delay = (ms) => {
  const start = Date.now();
  while (Date.now() - start < ms) {
    // no-op
  }
};
```

And do:
```js
document.getElementById('my-button').addEventListener('click', () => {
    delay(1000);

    console.log(":(")
  });
```

if `includeTimingsAsSpans` is true you should see INP timing spans like this:
![CleanShot 2024-07-31 at 13 06 39](https://github.com/user-attachments/assets/2b0a4423-783d-4aef-b486-73873268f998)
